src/main/java/com/google/cloud/tools/appengine/cloudsdk/internal/process/SynchronousOutputProcessRunner.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.appengine.cloudsdk.internal.process;
+
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessExitListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessOutputLineListener;
+import com.google.cloud.tools.appengine.cloudsdk.process.ProcessStartListener;
+import com.google.common.collect.Lists;
+
+/**
+ * A ProcessRunner that runs synchronous processes and makes their stderr and stdout streams
+ * available as Strings.
+ */
+public class SynchronousOutputProcessRunner extends DefaultProcessRunner implements ProcessRunner {
```

Comment:
I don't think it's redundant. It's a specific configuration of a `ProcessRunner`, that encapsulates a nontrivial configuration (see the Builder), and it adds behaviors that pertain to the synchronous execution of processes. If this class did not exist, all of this plumbing would have to take place in the `CloudSdk` class, which IMO should not need to take on the responsibility of knowing how to configure the ProcessRunner correctly, how to determine if the process exited successfully, etc. I feel pretty strongly that this class has value.

Ignoring the listeners set by the user is a different story. I agree that it's feels a bit weird, but I still think it's better than the alternative. A user of the CloudSdk would expect that the listeners that they configure are for piping the output from a long-running process somewhere. Think about this usage pattern:
```
ProcessOutputLineListener someListener = // define a listener that pipes output directly to the user
CloudSdk sdk = new CloudSdk.Builder().addStdOutLineListener(someListener).build();

if (!sdk.getComponents().contains(someRequiredComponent)) {
  // assume this code path is not followed
} else {
  sdk.runDevAppServerCommand("some command");
}
```
Assuming that the SynchronousOutputProcessRunner were to use the `someListener` that the CloudSdk was configured with, the user would see the output of the `gcloud components list` command that is called internally by `getComponents()`. I think that's bad.

By ignoring the user's configured output listeners, we can fully encapsulate the plumbing involved in finding the gcloud version. I think that's a very good thing, as it vastly simplifies the use of this method.